oracle/feed_stork.go

```diff
@@ -0,0 +1,260 @@
+package oracle
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/InjectiveLabs/metrics"
+	oracletypes "github.com/InjectiveLabs/sdk-go/chain/oracle/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/gorilla/websocket"
+	"github.com/pelletier/go-toml/v2"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	log "github.com/xlab/suplog"
+)
+
+var _ PricePuller = &storkPriceFeed{}
+
+type StorkFeedConfig struct {
+	ProviderName string `toml:"provider"`
+	Ticker       string `toml:"ticker"`
+	PullInterval string `toml:"pullInterval"`
+	Message      string `toml:"message"`
+	OracleType   string `toml:"oracleType"`
+}
+
+type storkPriceFeed struct {
+	providerName string
+	interval     time.Duration
+	message      string
+
+	logger  log.Logger
+	svcTags metrics.Tags
+
+	oracleType oracletypes.OracleType
+}
+
+func ParseStorkFeedConfig(body []byte) (*StorkFeedConfig, error) {
+	var config StorkFeedConfig
+	if err := toml.Unmarshal(body, &config); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal TOML config")
+		return nil, err
+	}
+
+	return &config, nil
+}
+
+// NewStorkPriceFeed returns price puller
+func NewStorkPriceFeed(cfg *StorkFeedConfig) (PricePuller, error) {
+	pullInterval := 1 * time.Minute
+	if len(cfg.PullInterval) > 0 {
+		interval, err := time.ParseDuration(cfg.PullInterval)
+		if err != nil {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (expected format: 60s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		if interval < 1*time.Second {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (minimum interval = 1s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		pullInterval = interval
+	}
+
+	var oracleType oracletypes.OracleType
+	if cfg.OracleType == "" {
+		oracleType = oracletypes.OracleType_PriceFeed
```

Comment:
shouldn't default to OracleType_Stork ?
```suggestion
		oracleType = oracletypes.OracleType_Stork
```

In fact, I'm confused why the StorkPriceFeed should be able to set any type other than "Stork", what's the rationale behind this? 

And moreover seeing that the type is hardcoded in the getter
```go
func (f *storkPriceFeed) OracleType() oracletypes.OracleType {
	return oracletypes.OracleType_Stork
}
```

Reply:
I set wrong for this oracle type, it should be `oracleType = oracletypes.OracleType_Stork` as you recommend